### ISSUE TITLE
Bug: --allow-eval does not work without --test or --run

### ISSUE THREAD

**Issue by umnikos** [ndmitchell/ghcid#293]
Code:
```haskell
addme = (+)

-- $> addme 2 2

main = print "hi"
```
Command: `ghcid --allow-eval ghcid-test.hs`
```
All good (1 module, at 16:40:03)
 
/home/umnikos/Programs/haskell/proofofconcept/ghcid-test.hs:3:1
$> addme 2 2
<interactive>:23:1-5: error:
    • Can't find interface-file declaration for variable addme
        Probable cause: bug in .hi-boot file, or inconsistent .hi file
        Use -ddump-if-trace to get an idea of which file caused the error
    • In the expression: addme 2 2
      In an equation for ‘it’: it = addme 2 2
```
Command: `ghcid --allow-eval --run ghcid-test.hs`
```
$> addme 2 2   
4              
Running test...
               
               
"hi"           
               
...done        
```

I did not see this mentioned in the readme, so I assume it is a bug. This is on version v0.8

**Comment by ndmitchell** [comment]
Thanks for the report. Yep, a bug. I've pushed a fix to HEAD - can you check it does what you were hoping?

**Comment by umnikos** [comment]
Yep, issue solved.

**Comment by ndmitchell** [comment]
Thanks, I've released 0.8.1 with the fix.
